landoapi/landing_worker.py

```diff
@@ -246,18 +246,37 @@ def run_job(self, job, repo, hgrepo, treestatus, patch_bucket):
             )
             return False
 
-        # Set the environment variable the `set_landing_system.py` hg
-        # extension will use to send the push user override.
-        os.environ["AUTOLAND_REQUEST_USER"] = job.requester_email
+        hgrepo.set_request_user(job.requester_email)
+        with hgrepo:
```

Comment:
This smells a bit like the context manager should be parameterized with the request user, maybe something like:
```
with hgrepo.with_user(job.requester_email):
    ...
```